src/ProtoSchema.ts

```diff
@@ -8,11 +8,20 @@ import {
 
 export default class ProtoSchema implements Schema {
   private message: Type
+  private root: protobuf.Root
 
-  constructor(schema: ConfluentSchema, opts?: ProtoOptions) {
+  constructor(schema: ConfluentSchema, opts?: ProtoOptions, references?: Schema[]) {
     const parsedMessage = protobuf.parse(schema.schema)
-    const root = parsedMessage.root
-    this.message = root.lookupType(this.getTypeName(parsedMessage, opts))
+    this.root = parsedMessage.root
+
+    if (references) {
+      const schemas = references as ProtoSchema[]
+      schemas.forEach(reference => {
+        this.root.add(reference.root)
```

Comment:
Need to double-check that it's safe to just `add()` the root here and that it does not assume ownership of the input argument. The safer way might be to use `addJSON(reference.root.toJSON().nested)`

Reply:
This is actually an issue, so I have updated the code to include a test that ensures we do not pollute the cache entry for imported schemas and to avoid that issue by cloning the reference here.